apps/editors/tests/test_views_themes.py

```diff
@@ -628,6 +627,41 @@ def test_single_rejected_reason_9_bug_1140346(self):
                                           args=[addon.slug]))
         eq_(res.status_code, 200)
 
+    @mock.patch('amo.utils.LocalFileStorage.delete')
+    def test_reject_theme_without_footer_bug_1142449(self, delete_mock):
+        """Don't fail on trying to deleted a non existent footer image."""
+        user = UserProfile.objects.get(email='persona_reviewer@mozilla.com')
+        self.login(user)
+        theme = self.theme_factory().persona
+        reject_url = reverse('editors.themes.commit')
+
+        form_data = amo.tests.formset(initial_count=1, total_count=1)
+        form_data['form-0-theme'] = str(theme.pk)
+        form_data['form-0-action'] = str(rvw.ACTION_REJECT)
+        form_data['form-0-reject_reason'] = 1
+
+        # With footer (RereviewQueueTheme created in theme_factory has one).
+        ThemeLock.objects.create(theme=theme, reviewer=user,
+                                 expiry=self.days_ago(-1))
+
+        with self.settings(ALLOW_SELF_REVIEWS=True):
+            res = self.client.post(reject_url, form_data)
+        eq_(res.status_code, 302)
+        assert delete_mock.call_count == 2  # Did try to delete the footer.
+
+        delete_mock.call_count = 0  # Clean slate.
+
+        # Without footer.
+        RereviewQueueTheme.objects.create(theme=theme,
+                                          header='pending_header.png')
+        ThemeLock.objects.create(theme=theme, reviewer=user,
+                                 expiry=self.days_ago(-1))
+
+        with self.settings(ALLOW_SELF_REVIEWS=True):
+            res = self.client.post(reject_url, form_data)
+        eq_(res.status_code, 302)
+        assert delete_mock.call_count == 1  # Didn't try to delete the footer.
```

Comment:
You should also try to look at the arguments to make sure what was being deleted.